packages/contracts-bedrock/contracts/L1/L2OutputOracle.sol

```diff
@@ -119,43 +116,36 @@ contract L2OutputOracle is Initializable, Semver {
 
         startingTimestamp = _startingTimestamp;
         startingBlockNumber = _startingBlockNumber;
-        latestBlockNumber = _startingBlockNumber;
     }
 
     /**
      * @notice Deletes all output proposals after and including the proposal that corresponds to
-     *         the given block number. Only the challenger address can delete outputs.
+     *         the given output index. Only the challenger address can delete outputs.
      *
-     * @param _l2BlockNumber L2 block number of the first output root to delete.
+     * @param _l2OutputIndex Index of the first L2 output to be deleted. All outputs after this
+     *                       output will also be deleted.
      */
     // solhint-disable-next-line ordering
-    function deleteL2Outputs(uint256 _l2BlockNumber) external {
+    function deleteL2Outputs(uint256 _l2OutputIndex) external {
         require(
             msg.sender == CHALLENGER,
             "L2OutputOracle: only the challenger address can delete outputs"
         );
 
-        // Simple check that accomplishes two things:
-        //   1. Prevents deleting anything before (and including) the starting block.
-        //   2. Prevents deleting anything other than a checkpoint block.
+        // Make sure we're not *increasing* the length of the array.
         require(
-            l2Outputs[_l2BlockNumber].outputRoot != bytes32(0),
-            "L2OutputOracle: cannot delete a non-existent output"
+            _l2OutputIndex < l2Outputs.length,
+            "L2OutputOracle: cannot delete outputs after the latest output index"
         );
 
-        // Prevent deleting beyond latest block number. Above check will miss this case if we
-        // already deleted an output and then the user tries to delete a later output.
-        require(
-            _l2BlockNumber <= latestBlockNumber,
-            "L2OutputOracle: cannot delete outputs after the latest block number"
-        );
+        uint256 prevNextL2OutputIndex = nextOutputIndex();
 
-        // We're setting the latest block number back to the checkpoint block before the given L2
-        // block number. Next proposal will overwrite the deleted output and following proposals
-        // will delete any outputs after that.
-        latestBlockNumber = _l2BlockNumber - SUBMISSION_INTERVAL;
+        // Use assembly to delete the array elements because Solidity doesn't allow it.
```

Comment:
Is this less deleting the elements & more setting the length of the array?

Reply:
Yes, but it has the same result